src/client.c

```diff
@@ -1016,18 +1016,32 @@ run_recipe_handler (gpointer user_data)
     RecipeData *recipe_data = (RecipeData*)user_data;
     AppData *app_data = recipe_data->app_data;
     const gchar **env = NULL;
+    gchar *command;
 
     // return the xml doc
     xmlBufferPtr buffer = xmlBufferCreate();
     gssize size = xmlNodeDump(buffer, app_data->xml_doc, recipe_data->recipe_node_ptr, 0, 1);
 
+    command = g_strdup_printf ("%s %s -- %s --stdin",
+                               app_data->rsh_cmd,
+                               recipe_data->connect_uri,
+                               app_data->restraint_path);
+
+    if (app_data->restraint_port != 0) {
```

Comment:
In my opinion I think it would have been simpler to do something to the effect:
If app_data->restraint_port !=0
    port_arg = g_strdup_printf(" --port ......
else
    port_arg = g_strdup_whatever(""); 
command = g_strdup_printf("%s %s -- %s --stdin %s", rsh_cmd, connect_uri, restraint_path, port_arg);

Reply:
I don't like to do `port_arg = g_strdup_whatever("");` when we know that we don't need the argument.
The branch can be simpler doing:
```
if (app_data->restraint_port != 0) {
    gchar *tmp_cmd;

    tmp_cmd = command;
    command = g_strdup_printf ("%s --port %d", command, app_data->restraint_port);
    g_free (tmp_cmd);
}
```
As we set `port = 0` by default, we can pass it even when user didn't provide it, and the outcome is the same, so I removed the if.

Reply:
That works too